js/util/metadata_to_json_schema.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Converts a metadata field's type to a JSON Schema type.
+ */
+function toJsonSchemaType(metadataFieldType) {
+  let type;
+
+  // This is sort of a reverse mapping of ui:widget to JSON Schema type
+  // https://github.com/mozilla-services/react-jsonschema-form#alternative-widgets
+  switch (metadataFieldType) {
+    case 'checkbox':
+      type = 'boolean';
+      break;
+    case 'file':
+    case 'tel':
+    case 'textarea':
+    default:
+      type = 'string';
+  }
+
+  return type;
+}
+
+/**
+ * Converts a single form field from the agency metadata to a JSON schema
+ * `properties` field.
+ */
+function toJsonSchemaProperty(metadataField) {
+  const type = toJsonSchemaType(metadataField.type);
+  const property = {
+    type,
+  };
+
+  // Copy over common schema properties
+  ['enum', 'enumNames'].forEach((key) => {
+    if (Object.prototype.hasOwnProperty.call(metadataField, key)) {
+      property[key] = metadataField[key];
+    }
+  });
+
+  return { [metadataField.name]: property };
+}
+
+/**
+ * Converts a single agency metadata form field to a uiSchema property to be
+ * used with react-jsonschema-form.
+ */
+function toUiSchemaProperty(metadataField) {
```

Comment:
Weren't we moving away from using uiSchema? Or only parts of it?

Reply:
The uiSchema is still necessary for react-jsonschema-form. They way they phrase it "JSON schema is the _what_ to render, uiSchema is the _how_ to render it".

But what I meant was: I think we should think of the uiSchema as an implementation detail and take it out of the metadata. I don't think folks consuming the metadata API are particularly interested in having uiSchema in it. We might put this in front of developers to validate this idea.